src/services/config.js

```diff
@@ -92,10 +101,11 @@ class ConfigData {
     let profile;
     if (this.projects.length > 0) {
       if (this.activeProfile) {
-        profile = this.projects.find((p) => p.id === this.activeProfile);
+        profile = this.getProfileFromConfigFileById(this.activeProfile);
       }
       if (!profile) {
         profile = this.projects[0];
+        this.addApiKeysToProject(profile);
       }
```

Comment:
I don't follow the logic here. If there's no active profile then pull the first from `projects` and add the keys to it from `profiles`?

Reply:
The idea is to continue to use the projects to read the first profile when active is not set until we are ready to remove it all. So we are just reading profile id from projects and then loading keys from profiles if possible

Reply:
Ah, then using the first `projects` makes sense, but there shouldn't be a scenario where an entry in `projects` has an API key in `profiles`.

Reply:
This ties to Anuj's PR where we update both projects(not keytar) and profiles when we create a new profile. Otherwise, referring to projects for the first profile when active is not set wouldn't work. However, our logic will now always first check profiles and fall back to keytar only when Keys are not present in the file

Reply:
I've left a comment on the other PR about where new/updated profiles should go. I think it's the cleanest path.